guides/source/action_text_overview.md

```diff
@@ -8,149 +8,145 @@ rich text content.
 
-[`rich_text_area`]: https://api.rubyonrails.org/classes/ActionView/Helpers/FormHelper.html#method-i-rich_text_area
+Since Action Text depends on polymorphic associations, which, in turn, involve storing class names in the database, it's crucial to keep the data in sync with the class names used in your Ruby code. This synchronization is essential to maintain consistency between the stored data and the class references in your codebase.
 
 ## Rendering Rich Text Content
 
-Instances of `ActionText::RichText` are capable of safely rendering themselves, so embed them directly into the page:
+Instances of `ActionText::RichText` can be directly embedded into a page because they have already sanitized their content for a safe render. You can display the content as follows:
 
 ```erb
-<%= @message.content %>
+<%= @article.content %>
 ```
 
-Learn more about Action Text's sanitization process in the documentation for the [`ActionText::RichText`](https://api.rubyonrails.org/classes/ActionText/RichText.html) class.
+`ActionText::RichText#to_s` safely transforms RichText into an HTML String. On the other hand `ActionText::RichText#to_plain_text` returns a string that is not HTML safe and should not be rendered in browsers. You can learn more about Action Text's sanitization process in the [documentation](https://api.rubyonrails.org/classes/ActionText/RichText.html) for the `ActionText::RichText` class.
 
-NOTE: If there's an attached resource within `content` field, it might not show properly unless you
-have *libvips/libvips42* package installed locally on your machine.
-Check their [install docs](https://www.libvips.org/install.html) on how to get it.
+NOTE: If there's an attached resource within `content` field, it might not show properly unless you have the necessary [dependencies for Active Storage](active_storage_overview.html#requirements) installed.
 
-By default, Action Text will render rich text content inside an element with the `.trix-content` class. Elements with this class, as well as the Action Text editor, are styled by the [`trix` stylesheet](https://unpkg.com/trix/dist/trix.css). To provide your own styles instead, remove `trix` from the pre-processor directives in the `app/assets/stylesheets/actiontext.css` file created by the installer:
+## Customizing the Rich Text Content Editor (Trix)
 
-```diff
- /*
-  * Provides a drop-in pointer for the default Trix stylesheet that will format the toolbar and
-  * the trix-editor content (whether displayed or under editing). Feel free to incorporate this
-  * inclusion directly in any other asset bundle and remove this file.
-  *
-- *= require trix
- */
+There may be times when you want to update the presentation of the editor to meet your stylistic requirements, this section guides on how to do that.
```

Comment:
Besides stylistic requirements, the styling may also need changing to accomodate for specific tooling, to workaround (perceived?) bugs in the Rails setup, or to support specific features:
- AFAIK, the provided ActionText and Trix CSS files do not support dark mode
- AFAIK, the setup is broken when adding ActionText to a Tailwind CSS app (*) 
- in an app using Tailwind CSS, it may make more sense to style rich text using the Typography plugin (`prose-*`)

I would therefore love to see more documentation/examples related to Tailwind CSS and dark mode.

<details>

In our app, we style using Tailwind CSS (incl. dark mode), so neither asset pipeline nor importmap are used to process and import CSS. Therefore we needed to remove the ActionText/Trix CSS files and instead use tailwind typography plugin utilities such as `prose md:prose-lg lg:prose-xl prose-headings:font-semibold prose-headings:mb-1 prose-h1:text-4xl prose-h2:text-3xl prose-h3:text-2xl dark:prose-invert max-w-3xl p-4 mx-auto [&_div]:mb-4`, but this takes some figuring out. This also does not answer the question of how to style the Trix editor itself using Tailwind (incl. in dark mode). The admin part of our app still uses sprockets/Sass for CSS so we haven't needed to resolve this question yet, but this is the next question on our list.

(*) According to my tests in a new Rails 7.1.3 app:
- create app: `rails new myapp --css=tailwind && cd myapp && rails g action_text:install && rails g scaffold article title:string body:rich_text && rails db:migrate`
- the above appends `@import 'actiontext.css';` to `application.tailwind.css` which seems incorrect (imports should be at the top of the file)
- the imported `actiontext.css` file contains a `*= require trix` sprockets directive which does not seem to be handled even after running `rails assets:precompile` and `bin/dev`. No styling gets applied in the Trix editor itself (e.g. h1 headers appear as plain text).

</details>

Reply:
What I do is to include the following in the application layout file. This works with Tailwind and with Propshaft and Sprockets without distinction.

```html+erb
    <%= stylesheet_link_tag 'trix', 'data-turbo-track': 'reload' %>
    <%= stylesheet_link_tag 'application', 'data-turbo-track': 'reload' %>>
    <%= stylesheet_link_tag 'tailwind', 'inter-font', 'data-turbo-track': 'reload' %>>
```

I put the other styles I have to emphasize that Trix should go at the beginning if you want to override the styles in application.css, for example.